### PR TITLE
Python goodies

### PR DESCRIPTION
**Pull request by jeremyrsmith**
- Automatically convert DataFrames to pyspark DataFrames on their way into the globals dict (no more `DataFrame(df, sqlContext)` needed, and they are marshalled with py4j instead of jep so they behave correctly!)
- Automatically convert pyspark DataFrames back to Scala DataFrames on their way into the result scope (they won't be `TypedPythonObject`s anymore!)
- Add streaming ops for Pandas DataFrames (I'm sure it's not perfect, but it's a start...)

### CODE REVIEW

File: polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala [hunk 1]
```diff
@@ -229,6 +261,10 @@ class PythonInterpreter private[python] (
       |        else:
       |            return node
       |
+      |def __polynote_mkindex__(l1, l2):
+      |    import pandas as pd
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala]: What if `pandas` is not present? Will we not get to this method?

Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala]: yeah, we should only get there if they have a pandas DataFrame (which would imply that pandas is available)

**Comment by jonathanindig** [comment]
Nice! But... maybe it might be good to add some tests? 😬

**Comment by jeremyrsmith** [comment]
@jindig added some tests

**Pull request merged**